### PR TITLE
Fix some performance issue in `VmMapping`

### PR DESCRIPTION
**Pull request by LclclcdIsACat**
This PR primarily addresses two issues:
- It checks if a page is mapped before unmap directly, eliminating the need for page walk on each address.
- The permission check and range check don't need to be done on a per-page basis but can be performed on the entire `VmMapping` as a whole.

### CODE REVIEW

File: kernel/aster-nix/src/vm/vmar/vm_mapping.rs [hunk 1]
```diff
@@ -590,17 +595,19 @@ impl VmMappingInner {
...
         if !self.perm.contains(*perm) {
             return_errno_with_message!(Errno::EACCES, "perm check fails");
         }
+        Ok(())
+    }
 
+    fn check_page_idx_range(&self, page_idx_range: &Range<usize>) -> Result<()> {
+        if page_idx_range.start * PAGE_SIZE < self.vmo_offset
+            || page_idx_range.end * PAGE_SIZE > self.vmo_offset + self.map_size
+        {
```
Comment (cchanging) [inline kernel/aster-nix/src/vm/vmar/vm_mapping.rs:608]: Should `page_idx_range.end * PAGE_SIZE` be `(page_idx_range.end + 1) * PAGE_SIZE`?

File: kernel/aster-nix/src/vm/vmar/vm_mapping.rs [hunk 2]
```diff
@@ -501,7 +504,9 @@ impl VmMappingInner {
             ..(range.end - map_to_addr + self.vmo_offset);
         let page_idx_range = get_page_idx_range(&vmo_map_range);
         for page_idx in page_idx_range {
-            self.unmap_one_page(vm_space, page_idx)?;
+            if self.mapped_pages.contains(&page_idx) {
+                self.unmap_one_page(vm_space, page_idx)?;
+            }
```
Comment (cchanging) [inline kernel/aster-nix/src/vm/vmar/vm_mapping.rs:509]: #726 will improved the efficiency here.

**Review by cchanging — COMMENTED** [review]
Good improvements for the permission instructions in `VmMapping`!

File: kernel/aster-nix/src/vm/vmar/vm_mapping.rs (hunk 1 above)
Comment (LclclcdIsACat) [reply kernel/aster-nix/src/vm/vmar/vm_mapping.rs:608]: No, `page_idx_range.end` is the next page after the valid pages.

File: kernel/aster-nix/src/vm/vmar/vm_mapping.rs (hunk 2 above)
Comment (LclclcdIsACat) [reply kernel/aster-nix/src/vm/vmar/vm_mapping.rs:509]: Oh, get it!

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**